lib/cc/engine/markdownlint.rb

```diff
@@ -7,33 +7,37 @@ module Engine
     class Markdownlint
       CONFIG_FILE = "./.mdlrc".freeze
       EXTENSIONS = %w[.markdown .md].freeze
+      UnexpectedMessageFormat = Class.new(StandardError)
 
-      def initialize(root, engine_config, io)
+      def initialize(root, engine_config, io, err_io = STDERR)
         @root = root
         @engine_config = engine_config
         @io = io
+        @err_io = err_io
         @contents = {}
       end
 
       def run
         return if include_paths.length == 0
 
-        pid, _, out, err = POSIX::Spawn.popen4("mdl", *mdl_options, *include_paths)
+        command = ["mdl", mdl_options, include_paths].flatten.join(" ")
```

Comment:
This is potentially problematic with path names that have spaces in them: you'all end up creating a command string referencing invalid paths. 

`Child.new` can take multiple args, similar to `popen4`. I think you could do `Child.new("mdl", *mdl_options, *include_paths)`.